hw/top_earlgrey/syn/chip_earlgrey_asic.sdc

```diff
@@ -546,49 +643,67 @@ set SPI_TPM_MOSI_IN_DEL_MAX [expr ${SPI_TPM_TCK_HALF} - ${SPI_TPM_MOSI_SETUP}]
 set SPI_TPM_MISO_OUT_DEL_MIN ${SPI_TPM_MISO_CLKQ_MIN}
 set SPI_TPM_MISO_OUT_DEL_MAX [expr ${SPI_TPM_TCK_HALF} - ${SPI_TPM_MISO_CLKQ_MAX}]
 
-create_clock -name SPI_TPM_CLK -add -period ${SPI_TPM_TCK} [get_ports ${SPI_DEV_CLK_PIN}]
+create_clock -name SPI_TPM_CLK -add -period ${SPI_TPM_TCK} [get_ports ${SPI_DEV_CLK_PIN}] -add
 set_clock_uncertainty ${SETUP_CLOCK_UNCERTAINTY} [get_clocks SPI_TPM_CLK]
 set_propagated_clock SPI_TPM_CLK
 
 create_generated_clock -name SPI_TPM_IN_CLK \
     -source [get_ports ${SPI_DEV_CLK_PIN}] -divide_by 1 \
     -master_clock SPI_TPM_CLK -add \
-    [get_pins top_earlgrey/u_spi_device/u_clk_spi_in_buf/clk_o]
+    [get_pins top_earlgrey/u_spi_device/u_clk_spi_in_buf/${CLK_PIN}]
 create_generated_clock -name SPI_TPM_OUT_CLK \
     -source [get_ports ${SPI_DEV_CLK_PIN}] -divide_by 1 -invert \
     -master_clock SPI_TPM_CLK -add \
-    [get_pins top_earlgrey/u_spi_device/u_clk_spi_out_buf/clk_o]
+    [get_pins top_earlgrey/u_spi_device/u_clk_spi_out_buf/${CLK_PIN}]
 
 # bidir ports
+if {$spec_constr} {
 set_input_delay -min ${SPI_TPM_MOSI_IN_DEL_MIN} [get_ports SPI_DEV_D0] \
     -clock_fall -clock SPI_TPM_CLK -add_delay
 set_input_delay -max ${SPI_TPM_MOSI_IN_DEL_MAX} [get_ports SPI_DEV_D0] \
     -clock_fall -clock SPI_TPM_CLK -add_delay
-
+} else {
+set_input_delay -min $spi_tpm_inp_min [get_ports SPI_DEV_D0] \
+    -clock_fall -clock SPI_TPM_CLK -add_delay
+set_input_delay -max $spi_tpm_inp_max [get_ports SPI_DEV_D0] \
+    -clock_fall -clock SPI_TPM_CLK -add_delay
+}
 # Half-cycle sampling has the host on the next rising edge.
+if {$spec_constr} {
 set_output_delay -min ${SPI_TPM_MISO_OUT_DEL_MIN} [get_ports SPI_DEV_D1] \
     -clock SPI_TPM_CLK -add_delay
 set_output_delay -max ${SPI_TPM_MISO_OUT_DEL_MAX} [get_ports SPI_DEV_D1] \
     -clock SPI_TPM_CLK -add_delay
-
+} else {
+set_output_delay -min $spi_tpm_out_val_min [get_ports SPI_DEV_D1] \
+    -clock SPI_TPM_CLK -add_delay
+set_output_delay -max $spi_tpm_out_val_max [get_ports SPI_DEV_D1] \
+    -clock SPI_TPM_CLK -add_delay
+}
 # SPI TPM CSB, the chip-select for TPM mode.
 # Any muxed port could be a SPI TPM CSB, but we only guarantee IOA7 meets
```

Comment:
Nit. This comment appears stale, given the addition of IOA2 below?

Reply:
I did have another look at this but could not find a reason why IOA2 should get these constraints. IOA2 doesn't seem to be related to SPI TPM at all. It's used a GPIO and in some cases we use it for SPI Host 1. But SPI Host 1 is not used for passthrough / TPM mode. This is only done for SPI device and SPI Host 0. So it's not clear to me why this is needed actually.

Reply:
The PD team got back to me about this: Some time in the past, IOA2 was determined to be used as chip select for quad SPI. During the tapeout, it was agreed that IOA7 should be the primary chip select target whereas IOA2 should remain as a secondary opportunistic target. I'll add a comment into the SDC to document this.